rclcpp/test/rclcpp/strategies/test_message_pool_memory_strategy.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+
+#include "gtest/gtest.h"
+
+#include "rclcpp/strategies/message_pool_memory_strategy.hpp"
+#include "test_msgs/msg/empty.hpp"
+
+using rclcpp::strategies::message_pool_memory_strategy::MessagePoolMemoryStrategy;
+
+class TestMessagePoolMemoryStrategy : public ::testing::Test
+{
+public:
+  void SetUp()
+  {
+    message_memory_strategy_ =
+      std::make_shared<MessagePoolMemoryStrategy<test_msgs::msg::Empty, 1>>();
+  }
+
+protected:
+  std::shared_ptr<MessagePoolMemoryStrategy<test_msgs::msg::Empty, 1>> message_memory_strategy_;
+};
+
+TEST_F(TestMessagePoolMemoryStrategy, construct_destruct) {
+}
```

Comment:
I don't typically like tests with no checks in them; they basically only test whether this compiles and doesn't throw an exception.  Also, if the tests are ever built with optimization enabled, the whole thing might be optimized away.  Can we at least check that the `message_memory_strategy_` pointer is not null, and maybe check something simple on the class?

(this might be made a bit easier by making `TestMessagePoolMemoryStrategy` have an "is-a" relationship with `MessagePoolMemoryStrategy`, but I'll leave it up to you whether you want to do that).

Reply:
> they basically only test whether this compiles and doesn't throw an exception

And segfaults! I find them pretty useful just when I'm debugging unit tests, but checking the invariance of the class on construction is a good habit I think. Since there are no EXPECT's in the SetUp, it makes sense to add some stuff here.

Added a few nullptr checks.